contracts/src/TeleporterTokenDestination.sol

```diff
@@ -87,40 +143,25 @@ abstract contract TeleporterTokenDestination is
      * - `amount` must be greater than 0
      * - `amount` must be greater than `input.primaryFee`
      */
-    function _send(SendTokensInput calldata input, uint256 amount) internal virtual {
-        require(
-            input.destinationBlockchainID != bytes32(0),
-            "TeleporterTokenDestination: zero destination blockchain ID"
-        );
-        require(
-            input.destinationBridgeAddress != address(0),
-            "TeleporterTokenDestination: zero destination bridge address"
-        );
+    function _send(SendTokensInput calldata input, uint256 amount) internal sendNonReentrant {
         require(input.recipient != address(0), "TeleporterTokenDestination: zero recipient address");
-
-        // Deposit the tokens sent from the user to the bridge,
-        // and set to adjusted amount after deposit.
-        amount = _deposit(amount);
-        require(
-            amount > input.primaryFee + input.secondaryFee,
-            "TeleporterTokenDestination: insufficient amount to cover fees"
+        require(input.requiredGasLimit != 0, "TeleporterTokenDestination: zero required gas limit");
```

Comment:
Nice catch. I changed it to >0 [here](https://github.com/ava-labs/teleporter-token-bridge/pull/84/files#diff-2eae3f817ea47c990b834a2e0e037db4c8d4ab2fc2973b180a9693664b19a272).

Even if relaying your own message, the required gas limit must be greater than 0 because other Teleporter will provide 0 gas to the message execution, which would automatically fail. Self-relayers can set 0 Teleporter fees, but they still have to set proper gas amounts 👍

Reply:
ah yes in `_tryExecuteMessage` makes sense